src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -733,4 +855,263 @@ export class NetworkAreaDiagramViewer {
+                }
+            }
+        }
+    }
+
+    private storeHalfEdgeAngle(
+        edgeNode: SVGGraphicsElement,
+        angleId: string,
+        sideIndex: number
+    ) {
+        if (!this.edgeAngles.has(angleId)) {
+            const halfEdge: HTMLElement = <HTMLElement>(
+                edgeNode.children.item(sideIndex)?.firstElementChild
+            );
+            if (halfEdge != null) {
+                const points = halfEdge.getAttribute('points');
+                if (points != null) {
+                    const angle = DiagramUtils.getPolylineAngle(points);
+                    if (angle != null) {
+                        this.edgeAngles.set(angleId, angle);
+                    }
+                }
+            }
+        }
+    }
+
+    private redrawVoltageLevelNode(
+        node: SVGGraphicsElement | null,
+        busNodeEdges: Map<string, SVGGraphicsElement[]>,
+        movedEdges: SVGGraphicsElement[] | null // list of moved edges, null = all edges
+    ) {
+        if (node != null) {
+            // get buses belonging to voltage level
+            const busNodes: NodeListOf<SVGGraphicsElement> =
+                this.container.querySelectorAll(
+                    'nad\\:busnode[vlnode="' + node.id + '"]'
+                );
+            // if single bus voltage level -> do not redraw anything
+            if (busNodes.length <= 1) {
+                return;
+            }
+            // sort buses by index
+            const sortedBusNodes: SVGGraphicsElement[] = [];
+            busNodes.forEach((busNode) => {
+                const index = busNode.getAttribute('index') ?? '-1';
+                if (+index >= 0) {
+                    sortedBusNodes[+index] = busNode;
+                }
+            });
+            const traversingBusEdgesAngles: number[] = [];
+            let redraw = false;
+            for (let index = 0; index < sortedBusNodes.length; index++) {
+                const busNode = sortedBusNodes[index];
+                // skip redrawing of first bus
+                if (index > 0 && redraw) {
+                    this.redrawBusNode(
+                        node,
+                        busNode,
+                        index,
+                        traversingBusEdgesAngles
+                    );
+                }
+                // add angles of edges starting from bus to traversing edges angles
+                const busEdges =
+                    busNodeEdges.get(busNode.getAttribute('svgid') ?? '-1') ??
+                    [];
+                busEdges.forEach((edge) => {
+                    const busId = busNode.getAttribute('svgid') ?? '-2';
+                    const busNode1 = edge.getAttribute('busnode1') ?? '-1';
+                    const edgeId = edge.getAttribute('svgid');
+                    const angleId =
+                        busId == busNode1 ? edgeId + '.1' : edgeId + '.2';
+                    const edgeAngle = this.edgeAngles.get(angleId);
```

Comment:
I think it would be clearer to call a `getEdgeAngle` method here, instead of computing it before in `redrawOtherVoltageLevelNode` and `moveEdge`. That's the only place where we need those angles.

This `getEdgeAngle` method would either return the value from the `edgeAngles` map or computes it and push it to the map. And if I'm not mistaken we have everything we need to compute them here.

Reply:
The angles could be needed to redraw the moved node, and possibly other connected nodes. 
For the moved node, the computation (and storing in the map) of the angles has been put in the moveEdge method because all the data to compute the angle (i.e. the 2 points of the edge polyline) are already available there (the points are computed to move the edge). The position of the code to compute and store the angle for the other edges follow the approach.
I can move all the computation of the angles here, but, at least for the edges connected to the moved node, it would require to retrieve the 2 polyline points used to compute the angle from the dom elements (as it is done when redrawing nodes connected to the moved node), instead of using already available values.
It would improve the readability of the code, loosing something in the performances.

Reply:
Ok, I understand.

In order not to deteriorate the performances but to improve the readability, I think we could do something in between
- set the values of the angle in advance only if the points are already known, that is, only in `moveEdge` function
- for the other angles, compute them here if `this.edgeAngles.get(angleId)` is undefined in `redrawVoltageLevelNode` function.
 
This will factorize the code calling `storeHalfEdgeAngle`.